crates/subspace-core-primitives/src/pieces.rs

```diff
@@ -218,6 +218,19 @@ impl PieceIndex {
         // Position is statically guaranteed to fit into u32
         (self.0 % ArchivedHistorySegment::NUM_PIECES as u64) as u32
     }
+
+    /// Is this piece index a source piece?
+    #[inline]
+    pub const fn is_source(&self) -> bool {
+        // With a replication factor of 2, even pieces are source pieces
+        self.0 % 2 == 0
+    }
+
+    /// Returns the next source piece index
+    #[inline]
+    pub const fn next_source_index(&self) -> PieceIndex {
+        PieceIndex(self.0.next_multiple_of(2))
```

Comment:
I'd probably tie it back to erasure coding here, though it results in the same value:
```suggestion
        PieceIndex(self.0.next_multiple_of(
            RecordedHistorySegment::ERASURE_CODING_RATE.1 / RecordedHistorySegment::ERASURE_CODING_RATE.0
        ))
```

Reply:
Happy to do that, I originally went with a hard-coded 2, because we do that in a lot of other places.

Changes in commit 0986e46025.